lsp/lsp.cpp

```diff
@@ -62,8 +73,12 @@ static const char *MethodNotFound       = "-32601";
 static const char *ServerNotInitialized = "-32002";
 //static const char *UnknownErrorCode     = "-32001";
 
+std::vector<Diagnostic> diagnostics;
+Runtime runtime(nullptr, 0, 4.0, 0);
 bool isInitialized = false;
+std::string rootUri = "";
 bool isShutDown = false;
+std::map<std::string, std::string> changedFiles;
```

Comment:
I'm not super thrilled that these are global variables. Could they perhaps be member variables of a class and then the methods which use them are member methods on that class?